pkg/tracegen/intake.go

```diff
@@ -0,0 +1,100 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package tracegen contains functions that generate a trace including transaction,
+// span, error, and logs using elastic APM Go agent and opentelemtry-go
+package tracegen
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"go.elastic.co/apm/v2"
+)
+
+type Config struct {
+	SampleRate float64
+	TraceID    apm.TraceID
+}
+
+// IndexIntakeV2Trace generate a trace including a transaction, a span and an error
+func IndexIntakeV2Trace(ctx context.Context, cfg Config, tracer *apm.Tracer) (apm.TraceID, error) {
```

Comment:
Since `Config` is already being used, what do you think about specifying the tracer in `Config`?

Reply:
I think the config will be later used by the otlp function (I will add endpoint secrets etc to the config in the other PR as it should be used by the otlp too.) so I didn't include the tracer in `Config` as I wanted to keep the boundary between the two

Reply:
I see, good point! I think we should keep it as is then.